### ISSUE TITLE
Licode RTSP stream problem 

### ISSUE THREAD

**Issue by ytjjyy** [ging/licode#464]
I have installed licode on Centos 7
./licode/scripts/installCentOsDeps.sh --enable-gpl
./licode/scripts/installErizo.sh
./licode/scripts/installNuve.sh
./licode/scripts/installBasicExample.sh

Then I use the demo basic_example, it works fine...

Next, I try to connect to RTSP stream (H264) and modify basic_example/public/script.js like this:  
var stream = Erizo.Stream({video: true, audio: false, url:"rtsp://user:pass@192.168.1.100/stream"});

    DEBUG:  Event: access-accepted
    DEBUG:  Creating URL from stream undefined
    ERROR:  Failed to get access to local media. Error was TypeError: Failed to execute 'createObjectURL' on 'URL': No function was found that matched the signature provided.

Who can help me solve this problem..thanks very much..

**Comment by ytjjyy** [comment]
@cracker0dks  SOS

**Comment by ytjjyy** [comment]
@lodoyun

**Comment by cracker0dks** [comment]
:D never tried that. So can't really help you with that. All I can say is, that this should work according to the doc. Are you sure your rtsp url is correct and running?

**Comment by ytjjyy** [comment]
@cracker0dks Yes,I have Tested wth vlc and correct，

Thanks very much,let me try again .

**Comment by cracker0dks** [comment]
Maybe check your transcoding of the stream. I dont know if you must habe a mkv container but I think so because you can only play mkv files aswell?!

**Comment by ytjjyy** [comment]
@cracker0dks 

Transcoding Info:

  bitrate=2000000
  framerate=25
  I frame num=100,IDR frame num=100
  H264 Max Delay: 1000ms
  H264 encode select: High Profile
  H264 encode select: High Quality Mode
  H264 encode standard: Level 4.0

**Comment by ytjjyy** [comment]
@cracker0dks 
Firefox debug info

    "DEBUG: " "Event: access-accepted" 
    "DEBUG: " "Creating URL from stream undefined" 
    "ERROR: " "Failed to get access to local media. Error was TypeError: Argument 1 is not valid for any      of the 2-argument overloads of URL.createObjectURL.." 
   "DEBUG: " "Event: access-denied"

**Comment by cracker0dks** [comment]
This transcoding info is from your rtsp server?

The error: "Failed to get access to local media" is the error if you dont allow the browser to access you mic/cam. Are you sure this is related to your rtsp stream?

**Comment by ytjjyy** [comment]
@cracker0dks 
![11111](https://cloud.githubusercontent.com/assets/12911640/15575654/90397430-2386-11e6-8168-f79f1453c709.png)

  
    DEBUG:  Event: access-accepted

I have granted access to local media and i get "Event: access-accepted" info   :(

**Comment by ytjjyy** [comment]
`var serverUrl = "/";
var localStream, room, recording, recordingId;

function getParameterByName(name) {
  name = name.replace(/[\[]/, "\\\[").replace(/[\]]/, "\\\]");
  var regex = new RegExp("[\\?&]" + name + "=([^&#]*)"),
      results = regex.exec(location.search);
  return results == null ? "" : decodeURIComponent(results[1].replace(/\+/g, " "));
}

function testConnection () {
  window.location = "/connection_test.html";
}

function startRecording () {
  if (room !== undefined){
    if (!recording){
      room.startRecording(localStream, function(id) {
        recording = true;
        recordingId = id;
      });
      
    } else {
      room.stopRecording(recordingId);
      recording = false;
    }
  }
}

window.onload = function () {
  recording = false;
  var screen = getParameterByName("screen");
  var config = {audio: false, video: true, 
	url:'rtsp://192.168.2.235:1516/stream',
data: true, screen: screen, videoSize: [640, 480, 640, 480]};
  // If we want screen sharing we have to put our Chrome extension id. The default one only works in our Lynckia test servers.
  // If we are not using chrome, the creation of the stream will fail regardless.
  if (screen){
    config.extensionId = "okeephmleflklcdebijnponpabbmmgeo";
  }
  localStream = Erizo.Stream(config);
  var createToken = function(userName, role, callback) {

    var req = new XMLHttpRequest();
    var url = serverUrl + 'createToken/';
    var body = {username: userName, role: role};

    req.onreadystatechange = function () {
      if (req.readyState === 4) {
        callback(req.responseText);
      }
    };

    req.open('POST', url, true);
    req.setRequestHeader('Content-Type', 'application/json');
    req.send(JSON.stringify(body));
  };

  createToken("user", "presenter", function (response) {
    var token = response;
    console.log(token);
    room = Erizo.Room({token: token});

    localStream.addEventListener("access-accepted", function () {
      var subscribeToStreams = function (streams) {
        for (var index in streams) {
          var stream = streams[index];
          if (localStream.getID() !== stream.getID()) {
            room.subscribe(stream);
            stream.addEventListener("bandwidth-alert", function (evt){
                console.log("Bandwidth Alert", evt.msg, evt.bandwidth);
            });
            

          }
        }
      };

      room.addEventListener("room-connected", function (roomEvent) {

        room.publish(localStream, {maxVideoBW: 300}); 
        subscribeToStreams(roomEvent.streams);
      });

      room.addEventListener("stream-subscribed", function(streamEvent) {
        var stream = streamEvent.stream;
        var div = document.createElement('div');
        div.setAttribute("style", "width: 320px; height: 240px;");
        div.setAttribute("id", "test" + stream.getID());

        document.body.appendChild(div);
        stream.show("test" + stream.getID());

      });

      room.addEventListener("stream-added", function (streamEvent) {
        var streams = [];
        streams.push(streamEvent.stream);
        subscribeToStreams(streams);
        document.getElementById("recordButton").disabled = false;
      });

      room.addEventListener("stream-removed", function (streamEvent) {
        // Remove stream from DOM
        var stream = streamEvent.stream;
        if (stream.elementID !== undefined) {
          var element = document.getElementById(stream.elementID);
          document.body.removeChild(element);
        }
      });
      
      room.addEventListener("stream-failed", function (streamEvent){
          console.log("Stream Failed, act accordingly");
      });

      room.connect();

      localStream.show("myVideo");

    });
    localStream.init();
  });
};`

**Comment by ytjjyy** [comment]
@cracker0dks 

Here's the media streaming part,can you give some advice.

`

var localStream;

window.onload = function () {

var config = {
	video: true, 
	audio: false,
 	url:"rtsp://192.168.2.201:554/stream"
	};


  	localStream = Erizo.Stream(config);

    	localStream.addEventListener("access-accepted", function ()
	{
		localStream.show("myVideo");

    	});

    localStream.init();
};`

**Comment by cracker0dks** [comment]
```
var CONFIG = {audio: false, video: true, data: false, url: 'rtsp://192.168.2.201:554/stream'};
var localStream = Erizo.Stream(CONFIG);
localStream.addEventListener('access-accepted', function () {
    Room.publish(localStream);
});
localStream.init();
```
what about this?
Are you sure you have no audio in your stream ?

**Comment by ytjjyy** [comment]
@cracker0dks 
Im sure.
Im so depressed :(
Help me pls.

**Comment by cracker0dks** [comment]
you have to remove the `localStream.show("my_local_video");` line frome your example because you cant display a rtsp stream in a video element out of the box. Than your error is gone.

**Comment by ytjjyy** [comment]
@cracker0dks 
Thanks very much.You are good man.The first error is work out.

Now, which format is can be accepted by licode :)

The video displayed on screen is green.
![2323](https://cloud.githubusercontent.com/assets/12911640/15625791/f6cfe5ce-24e6-11e6-9e50-0587966e7db6.png)

**Comment by ytjjyy** [comment]
@cracker0dks 
I have solved this problem by reset RTSP server output params.

```
enable video sclaing
video sclaing width=640
video sclaing height=480
```
Also you can modify '/ging/licode/blob/master/erizo/src/erizo/media/ExternalInput.cpp' line 133 and 134.
The height and width of the stream transcoded should be the same with the original stream.

Now it works fine in firefox, but Chrome is still black screen :(
